ios/brave-ios/Sources/AIChat/Subscription/SDK/BraveStoreSDK.swift

```diff
@@ -0,0 +1,412 @@
+      // There is currently no subscription so check if there was a restored receipt
+      if Bundle.main.appStoreReceiptURL?.lastPathComponent == "sandboxReceipt" {
+        return .sandbox
+      }
+      
+      // If the application has a Parent Process-ID then it is being debugged
+      if getppid() != 1 {
+        return .xcode
+      }
+      
+      return .production
+    }
+    
+    // Retrieve the current environment from StoreKit's Renewal Information
+    switch renewalInfo {
+    case .verified(let renewalInfo), .unverified(let renewalInfo, _):
+      if #available(iOS 16.0, *) {
+        return .init(rawValue: renewalInfo.environment.rawValue) ?? .production
+      }
+      
+      return .init(rawValue: renewalInfo.environmentStringRepresentation) ?? .production
+    }
+  }
+  
+  /// A boolean indicating whether or not all the VPN product offerings has been loaded
+  public var isVpnProductsLoaded: Bool {
+    if vpnMonthlyProduct != nil || vpnYearlyProduct != nil {
+      return true
+    }
+    
+    return false
+  }
+  
+  /// A boolean indicating whether or not all the Leo product offerings has been loaded
+  public var isLeoProductsLoaded: Bool {
+    if leoMonthlyProduct != nil || leoYearlyProduct != nil {
+      return true
+    }
+    
+    return false
+  }
+  
+  /// Refreshes all the Skus SDK orders
+  public func refreshAllSkusOrders() {
+    
+  }
+  
+  /// Restores a single purchased product
+  /// - Parameter product: The product whose purchase receipt to restore
+  /// - Returns: Returns true if receipt restoration was successful. False otherwise
+  public func restorePurchase(_ product: BraveStoreProduct) async -> Bool {
+    if await currentTransaction(for: product) != nil {
+      try? await AppStoreReceipt.sync()
+      return true
+    }
+    
+    return false
+  }
+  
+  /// Restores all purchased products
+  /// - Returns: Returns true if receipt restoration was successful. False otherwise
+  @MainActor
+  public func restorePurchases() async -> Bool {
+    #if STOREKIT2
+    var didRestore = false
+    for await result in Transaction.currentEntitlements {
+      if case .verified(let transaction) = result {
+        if let product = BraveStoreProduct(rawValue: transaction.productID) {
+          // Update subscription status for the product
+          switch product {
+          case .vpnMonthly, .vpnYearly:
+            vpnSubscriptionStatus = await transaction.subscriptionStatus
+          case .leoMonthly, .leoYearly:
+            leoSubscriptionStatus = await transaction.subscriptionStatus
+          }
+          
+          // Update SkusSDK
+          do {
+            try await self.updateSkusPurchaseState(for: product)
+            didRestore = true
+          } catch {
+            Logger.module.error("[BraveStoreSDK] - Failed to restore purchased product receipt: \(error)")
```

Comment:
nit: You shouldn't prefix in the logger like this, you should create a new `Logger` with `BraveStoreSDK` as the category. Feel free to make custom Logger's per class even.